packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianReferences.test.js

```diff
@@ -54,60 +52,62 @@ const references = [
   },
 ];
 
-describe('<AccordianReferences>', () => {
-  let wrapper;
+const link = { href: 'link' };
 
+const setup = (propOverrides) => {
   const props = {
     compact: false,
     data: references,
     highContrast: false,
     isOpen: false,
     onToggle: jest.fn(),
-    createFocusSpanLink: jest.fn(),
+    createFocusSpanLink: () => link,
+    ...propOverrides,
   };
 
-  beforeEach(() => {
-    wrapper = shallow(<AccordianReferences {...props} />);
-  });
+  return render(<AccordianReferences {...props} />);
+};
 
+describe('AccordianReferences tests', () => {
   it('renders without exploding', () => {
-    expect(wrapper).toBeDefined();
-    expect(wrapper.exists()).toBe(true);
+    expect(() => setup()).not.toThrow();
+  });
+
+  it('renders the correct number of references', () => {
+    setup();
+
+    expect(screen.getByRole('switch', { name: 'References (3)' })).toBeInTheDocument();
   });
 
   it('renders the content when it is expanded', () => {
-    wrapper.setProps({ isOpen: true });
-    const content = wrapper.find(References);
-    expect(content.length).toBe(1);
-    expect(content.prop('data')).toBe(references);
+    setup({ isOpen: true });
+
+    expect(screen.getByRole('switch', { name: 'References (3)' })).toBeInTheDocument();
+    expect(screen.getAllByRole('link', { name: /^service\d\sop\d/ })).toHaveLength(2);
+    expect(screen.getByRole('link', { name: /^View\sLinked/ })).toBeInTheDocument();
   });
 });
 
-describe('<References>', () => {
-  let wrapper;
+describe('References tests', () => {
+  const setup = (propOverrides) => {
+    const props = {
+      data: references,
+      createFocusSpanLink: () => link,
+      ...propOverrides,
+    };
 
-  const props = {
-    data: references,
-    createFocusSpanLink: jest.fn(),
+    return render(<References {...props} />);
   };
 
-  beforeEach(() => {
-    wrapper = shallow(<References {...props} />);
+  it('renders without exploding', () => {
+    expect(() => setup()).not.toThrow();
   });
 
   it('render references list', () => {
-    const refLinks = wrapper.find(ReferenceLink);
-    expect(refLinks.length).toBe(references.length);
-    refLinks.forEach((refLink, i) => {
-      const span = references[i].span;
-      const serviceName = refLink.find('span.span-svc-name').text();
-      if (span && span.traceID === traceID) {
-        const endpointName = refLink.find('small.endpoint-name').text();
-        expect(serviceName).toBe(span.process.serviceName);
-        expect(endpointName).toBe(span.operationName);
-      } else {
-        expect(serviceName).toBe('View Linked Span ');
-      }
-    });
+    setup();
+
+    expect(screen.queryByRole('switch')).not.toBeInTheDocument();
+    expect(screen.getAllByRole('link', { name: /^service\d\sop\d/ })).toHaveLength(2);
+    expect(screen.getByRole('link', { name: /^View\sLinked/ })).toBeInTheDocument();
```

Comment:
think this entire describe block is pretty redundant now since the previous describe block is already testing this behaviour.

Reply:
Done :)